app/src/main/java/fr/free/nrw/commons/customselector/ui/selector/CustomSelectorActivity.kt

```diff
@@ -156,11 +279,26 @@ class CustomSelectorActivity: BaseActivity(), FolderClickListener, ImageSelectLi
     }
 
     /**
-     * override Selected Images Change, update view model selected images.
+     * override Selected Images Change, update view model selected images and change UI.
      */
-    override fun onSelectedImagesChanged(selectedImages: ArrayList<Image>) {
+    override fun onSelectedImagesChanged(selectedImages: ArrayList<Image>,
+                                         selectedNotForUploadImages: Int) {
         viewModel.selectedImages.value = selectedImages
 
+        if (selectedNotForUploadImages > 0) {
+            upload.isEnabled = false
+            upload.alpha = 0.5f
+        } else {
+            upload.isEnabled = true
+            upload.alpha = 1f
+        }
+
+        not_for_upload.text =
+            if (selectedImages.size == selectedNotForUploadImages)
```

Comment:
Conditional blocks would be better, while this looks and works well, we need to follow conventions.